### ISSUE TITLE
Fix a11y issues with Tab-like Nav

### ISSUE THREAD

**Issue by akegan** [bloom-housing/bloom#1489]
The Tab-like Nav component raises the error "Nested interactive controls are not announced by screen readers" in the automated a11y tests implemented in #1450.

We temporarily disabled the rule to get the tests passing, but we should follow up to re-assess how the tabs are setup. 

More info about the rule can be found [on the deque website](https://dequeuniversity.com/rules/axe/4.2/nested-interactive?application=axe-puppeteer)

Full output from the test
```
[
  {
    "description": "Nested interactive controls are not announced by screen readers",
    "help": "Ensure interactive controls are not nested",
    "helpUrl": "https://dequeuniversity.com/rules/axe/4.2/nested-interactive?application=axe-puppeteer",
    "id": "nested-interactive",
    "impact": "serious",
    "nodes": [
      {
        "all": [],
        "any": [
          {
            "data": null,
            "id": "no-focusable-content",
            "impact": "serious",
            "message": "Element has focusable descendants",
            "relatedNodes": []
          }
        ],
        "failureSummary": "Fix any of the following:
      Element has focusable descendants", "html": "<li role=\"tab\" aria-selected=\"true\" class=\"tab-nav__tab\"><a class=\"is-active\" href=\"/other\" aria-current=\"page\">Other</a></li>",
        "impact": "serious",
        "none": [],
        "target": [
          ".tab-nav__tab[role=\"tab\"]:nth-child(1)"
        ]
      },
      {
        "all": [],
        "any": [
          {
            "data": null,
            "id": "no-focusable-content",
            "impact": "serious",
            "message": "Element has focusable descendants",
            "relatedNodes": []
          }
        ],
        "failureSummary": "Fix any of the following:
      Element has focusable descendants", "html": "<li role=\"tab\" class=\"tab-nav__tab\"><a href=\"/default\">Default<span class=\"tag is-normal is-pill\">15</span></a></li>",
        "impact": "serious",
        "none": [],
        "target": [
          ".tab-nav__tab[role=\"tab\"]:nth-child(2)"
        ]
      }
    ],
    "tags": [
      "cat.keyboard",
      "wcag2a",
      "wcag412"
    ]
  }
]
```
